front/lib/api/assistant/configuration.ts

```diff
@@ -146,23 +157,28 @@ export async function getAgentConfigurations<V extends "light" | "full">({
     }
   })();
 
-  let globalAgentsPromise = getGlobalAgents(auth, globalAgentIdsToFetch).then(
-    (globals) =>
-      globals.filter(
-        (a) =>
-          !agentPrefix ||
-          a.name.toLowerCase().startsWith(agentPrefix.toLowerCase())
-      )
+  const applySortAndLimit = (results: AgentConfigurationType[]) => {
+    const sortStrategy = sort && sortStrategies[sort];
+
+    const sortedResults = sortStrategy ? results.sort(sortStrategy) : results;
+
+    return limit ? sortedResults.slice(0, limit) : sortedResults;
+  };
+
+  const allGlobalAgents = await getGlobalAgents(auth, globalAgentIdsToFetch);
```

Comment:
I remember now why there was a promise, it was to fasten the total time of the call to getAgentConfs by paralelizing IO
Can we find a way to preserve that?